klippy/extras/safe_power.py

```diff
@@ -0,0 +1,116 @@
+# Stepper Driver Anti-SNAFU Protection support
+#
+# Copyright (C) 2021  Lars R. Hansen <popshansen@hotmail.com>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+import logging, time
+
+class BackwardError(Exception):
```

Comment:
Raising a BackwardError will effectively halt Klipper on an error.  I think it would be preferable for the code to raise a `printer.config_error()` or `printer.command_error` on a problem.  That provides better diagnostics.

Reply:
I had my thought on execption types. As I understand the config_error is issues in printer.cfg. But I can change the code to use it. If it raise an error you need to switch off you printer and rotate the stepper drive to the correct orientation. Maybe the switch off printer message should be part of the error message, so the user know's what todo to resolve the issue.

/Lars